craco.config.js

```diff
@@ -1,71 +1,99 @@
         https: require.resolve("https-browserify"), // required for asyncapi parser
         stream: require.resolve("stream-browserify"),
         util: require.resolve("util"),
         zlib: false,
-      },
-      alias: {
+      };
+
+      webpackConfig.resolve.alias = {
+        ...webpackConfig.resolve.alias,
         // This alias doesn't pull any languages into bundles and works as monaco-editor-core was installed
         "monaco-editor$": "monaco-editor/esm/vs/editor/edcore.main.js",
         // This alias make sure we don't pull two different versions of monaco-editor
         "monaco-editor": "/node_modules/monaco-editor",
-      },
-    },
-    plugins: [
-      new webpack.ProvidePlugin({
-        process: "process/browser.js",
-        Buffer: ["buffer", "Buffer"],
-      }),
-      new NodePolyfillPlugin(),
-    ],
-    module: {
-      rules: [
-        {
-          test: /\.css$/,
-          use: ["style-loader", "css-loader"],
-        },
-        /**
-         * The default way in which webpack loads wasm files won’t work in a worker,
-         * so we will have to disable webpack’s default handling of wasm files and
-         * then fetch the wasm file by using the file path that we get using file-loader.
-         *
-         * Resource: https://pspdfkit.com/blog/2020/webassembly-in-a-web-worker/
-         *
-         * Alternatively, WASM file can be bundled directly into JavaScript bundle as data URLs.
-         * This configuration reduces the complexity of WASM file loading
-         * but increases the overal bundle size:
-         *
-         * {
-         *   test: /\.wasm$/,
-         *   type: 'asset/inline',
-         * }
-         */
-        {
-          test: /\.wasm$/,
-          loader: "file-loader",
-          type: "javascript/auto", // this disables webpacks default handling of wasm
-        },
-      ],
+      };
+
+      webpackConfig.module.rules.push(
+          /**
+           * The default way in which webpack loads wasm files won’t work in a worker,
+           * so we will have to disable webpack’s default handling of wasm files and
+           * then fetch the wasm file by using the file path that we get using file-loader.
+           *
+           * Resource: https://pspdfkit.com/blog/2020/webassembly-in-a-web-worker/
+           *
+           * Alternatively, WASM file can be bundled directly into JavaScript bundle as data URLs.
+           * This configuration reduces the complexity of WASM file loading
+           * but increases the overal bundle size:
+           *
+           * {
+           *   test: /\.wasm$/,
+           *   type: 'asset/inline',
+           * }
+           */
+          {
+            test: /\.wasm$/,
+            loader: "file-loader",
+            type: "javascript/auto", // this disables webpacks default handling of wasm
+          },
+      );
+
+      if (shouldUseSourceMap) {
+        webpackConfig.module.rules[0].exclude = [
```

Comment:
SwaggerEditor uses some dependencies which incorrect source map references. This prevents webpack to emit warnings when those incorrect source maps are intercepted.